### DISCUSSION TITLE
Form actions

### DISCUSSION THREAD

**Discussion by joshshipton** [sveltejs/kit#11146] [Q&A]
Is there a way to check for successful form submission and then return that to the client side after submitting a form that uses a form action. E.g form is submitted, form action on server runs, then the client side gets the response. Pretty much just want to be able to discern whether or not there was an error submitting the form on the server to the client. thanks in advance ^_^

**Comment by hjaber** [comment]
https://kit.svelte.dev/docs/form-actions#anatomy-of-an-action

```svelte
<script lang="ts">
	import type { PageData, ActionData } from './$types';
	
	export let data: PageData;
	
	export let form: ActionData;
</script>

{#if form?.success}
	<!-- this message is ephemeral; it exists because the page was rendered in
		   response to a form submission. it will vanish if the user reloads -->
	<p>Successfully logged in! Welcome back, {data.user.name}</p>
{/if}
```

you could also use `form?.success` if you needed to in the `<script>`  for logic

**Comment by joshshipton** [comment]
This makes sense but if i want to redirect after form submission how can I make this message persist through page navigation through svelte-stores?
